### DISCUSSION TITLE
Data importer : Could not load the users currency preference.

### DISCUSSION THREAD

**Discussion by Pyco** [firefly-iii/firefly-iii#8944] [Firefly III] [answered]
Hello there,

I can't import my data through CSV file since a few days because of this : 
```
[2024-06-02 14:16:38] local.ERROR: Error accessing "https://my.domain/firefly/api/v1/preferences/currencyPreference?". Status code is 500. Body is: {"message":"Internal Firefly III Exception: Please use api\/v1\/currencies\/default instead.","exception":"FireflyIII\\Exceptions\\FireflyException"}
[2024-06-02 14:16:38] local.ERROR: Could not load the users currency preference. {"exception":"[object] (App\\Exceptions\\ImporterErrorException(code: 0): Could not load the users currency preference. at /var/www/firefly-iii-di/app/Services/CSV/Conversion/Routine/PseudoTransactionProcessor.php:110)
```

It seems it can't get the currency preference from the main FireflyIII instance, but I could connect the data importer as usual. My token / OAuth client seems ok.
The error shows when I start the job of conversion.
Firefly III version : 6.1.16

Thanks for your help :)

**Comment by JC5** [comment]
Make sure you run the latest version of the data importer :)

**Reply by Pyco** [reply]
I run it through a Yunohost package, but ... it stuck in 1.2.2, thought I was up-to-date, sorry.
I don't know how to update it manually though, and it seems no one wants to update this package since last year at least.
Thank you for your quick answer and nice work, I really like this application.

**Comment by Pyco** [comment]
I tried something by unzipping the last release into my yunohost data importer folder.
My server was not on php8.3 by default so I changed it.

But now I have this issue : Composer detected issues in your platform: Your Composer dependencies require a PHP version ">= 8.3.0". 

I tried to check my nginx conf, it appears that I have 
`fastcgi_pass unix:/var/run/php/php8.2-fpm-firefly-iii-di.sock;`
The 1 million question, how do I generate .sock file in php8.3 for the data importer ? Is there any update to perform somewhere ?
For FireflyIII I have the 8.3 .stock file.

**Reply by Pyco** [reply, marked as answer]
Ah  !
I did it, I moved my php-fpm conf for the data importer ("firefly-iii-di" in my example) :
`sudo mv /etc/php/8.2/fpm/pool.d/firefly-iii-di.conf /etc/php/8.3/fpm/pool.d/`

Then I edited the firefly-iii-di.conf to change from the 8.2 to 8.3 socket : 
`listen = /var/run/php/php8.3-fpm-firefly-iii-di.sock`

Then I restarted php-fpm service, then change the nginx conf in : 
`/etc/nginx/conf.d/my-domain/firefly-iii-di.conf`
To
`fastcgi_pass unix:/var/run/php/php8.3-fpm-firefly-iii-di.sock;`

And restart nginx service.

I had other issues with the packaging, but it's maybe not interesting here :)

**Comment by codemicro** [comment]
For anyone else that stumbles across this issue, make sure you're using the most up-to-date version of the importer!

I was running into this issue because an endpoint the version of the importer that I was running was trying to call a deprecated endpoint in Firefly and balking at the error returned.
